airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/check/DestinationCheckOperation.kt

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.check
+
+import io.airbyte.cdk.command.DestinationConfiguration
+
+/**
+ * A check operation that is run before the destination is used.
+ *
+ * * Implementors must provide a [check] method that validates the configuration.
+ * * Implementors may provide a [cleanup] method that is run after the check is complete.
+ * * [check] should throw an exception if the configuration is invalid.
+ * * [cleanup] should not throw exceptions.
+ * * Implementors should not perform any side effects in the constructor.
+ * * Implementors should not throw exceptions in the constructor.
+ * * Implementors should not inject configuration; only use the config passed in [check].
+ */
+interface DestinationCheckOperation<C : DestinationConfiguration> {
```

Comment:
I get what this is, but IMHO it's worth thinking of a better name.

Reply:
I'm open to suggestions. :)

Reply:
Naming things is hard!
`DestinationChecker`?